### PR TITLE
Add custom analytics data provider sample

### PR DESCRIPTION
**Pull request by YasasRangika**
## Purpose
To guide folks to add custom analytics data to the existing event schema.

## Goals
Example for: https://github.com/wso2/api-manager/issues/443

## Approach
Follow the instructions provided in README

### CODE REVIEW

File: analytics-custom-data-provider/pom.xml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+<!--        Below versions can be found by searching the component name in <APIM_HOME>/repository/components/plugins directory.-->
+<!--        org.wso2.carbon.apimgt.gateway component version-->
+        <carbon.apimgt.gateway.version>9.0.174.182</carbon.apimgt.gateway.version>
+<!--        org.wso2.carbon.apimgt.common.analytics component version-->
+        <carbon.apimgt.ommon.analytics.version>9.0.174.182</carbon.apimgt.ommon.analytics.version>
+<!--        synapse-core version-->
+        <synapse.version>2.1.7-wso2v227</synapse.version>
+    </properties>
+
+</project>
```
Comment (ruks) [inline analytics-custom-data-provider/pom.xml:99]: new line

File: analytics-custom-data-provider/src/main/java/org/wso2/carbon/apimgt/gateway/sample/publisher/CustomDataProvider.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package org.wso2.carbon.apimgt.gateway.sample.publisher;
```
Comment (ruks) [inline analytics-custom-data-provider/src/main/java/org/wso2/carbon/apimgt/gateway/sample/publisher/CustomDataProvider.java:1]: licence file

**Review by ruks — APPROVED** [review]

File: analytics-custom-data-provider/pom.xml (hunk 1 above)
Comment (YasasRangika) [reply analytics-custom-data-provider/pom.xml]: Fixed with https://github.com/wso2/samples-apim/pull/94/commits/9bd439c27f8ccbe28d1973bdacff8df9d7c4a7ab

File: analytics-custom-data-provider/src/main/java/org/wso2/carbon/apimgt/gateway/sample/publisher/CustomDataProvider.java (hunk 2 above)
Comment (YasasRangika) [reply analytics-custom-data-provider/src/main/java/org/wso2/carbon/apimgt/gateway/sample/publisher/CustomDataProvider.java:18]: Fixed with https://github.com/wso2/samples-apim/pull/94/commits/9bd439c27f8ccbe28d1973bdacff8df9d7c4a7ab

**Pull request merged**